src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/ExecutorFactory.cs

```diff
@@ -11,7 +11,7 @@ namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
 {
     public static class ExecutorFactory
     {
-        public static Func<Page, object, Task<IActionResult>> CreateExecutor(
+        public static Func<Page, object, object[], Task<IActionResult>> CreateExecutor(
```

Comment:
Don't we need an `IActionResult` so we can run it's `ExecuteResultAsync` method after we execute it [here](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvoker.cs#L386)?